src/jdk.crypto.ec/share/classes/sun/security/ec/SunEC.java

```diff
@@ -182,7 +182,7 @@ public Object newInstance(Object ctrParamObj)
                     }
                 } else  if (type.equals("KeyAgreement")) {
                     if (algo.equals("ECDH")) {
-                        if (useNativeEC) {
+                        if (useNativeEC && NativeCrypto.isLoaded()) {
```

Comment:
A similar change should be made on line 351.

Reply:
L351 can proceed w/ `useNativeEC` enabled, `NativeCrypto.isLoaded()` is needed only when a `NativeCripto` usage is imminent.
Invoking `NativeCrypto.isLoaded()` at `SunEC` instantiation stage caused a similar issue like https://github.com/OpenLiberty/open-liberty/issues/23360 before.